advanced_functionality/distributed_tensorflow_mask_rcnn/mask-rcnn-s3.ipynb

```diff
@@ -0,0 +1,566 @@
+    "\n",
+    "We run the training job in your private VPC, so we need to set the ```subnets``` and ```security_group_ids``` prior to running the cell below. You may specify multiple subnet ids in the ```subnets``` list. The subnets included in the ```sunbets``` list must be part of the output of  ```./stack-sm.sh``` CloudFormation stack script used to create this notebook instance. Specify only one security group id in ```security_group_ids``` list. The security group id must be part of the output of  ```./stack-sm.sh``` script."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "security_group_ids =  # ['sg-xxxxxxxx']\n",
+    "subnets =      # [ 'subnet-xxxxxxx']\n",
+    "sagemaker_session = sagemaker.session.Session(boto_session=session)\n",
+    "obj_detection = Estimator(training_image,\n",
+    "                                         role, \n",
+    "                                         train_instance_count=4, \n",
+    "                                         train_instance_type='ml.p3.16xlarge',\n",
+    "                                         train_volume_size = 100,\n",
+    "                                         train_max_run = 400000,\n",
+    "                                         input_mode= 'File',\n",
+    "                                         output_path=s3_output_location,\n",
+    "                                         sagemaker_session=sagemaker_session, \n",
+    "                                         hyperparameters = hyperparameters,\n",
+    "                                         metric_definitions = metric_definitions,\n",
+    "                                         base_job_name=\"mask-rcnn-s3\",\n",
+    "                                         subnets=subnets,\n",
+    "                                         security_group_ids=security_group_ids)\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Finally, we launch the SageMaker training job. \n",
+    "\n",
+    "The estimated time for downloading data to all the training instances is 20 minutes. The time to complete the training depends on type and number of training instances, and the training image used for training."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "obj_detection.fit(inputs=data_channels, logs=True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  }
+ ],
+ "metadata": {
+  "kernelspec": {
+   "display_name": "conda_amazonei_mxnet_p36",
+   "language": "python",
+   "name": "conda_amazonei_mxnet_p36"
```

Comment:
Minor note on the kernelspec for these notebooks. I believe the `amazonei` version of the kernel is not relevant here as Elastic Inference Adapters are not involve. In case there are any compatibility/staleness problems, I think it's best to keep it simple using the `conda_tensorflow_p36` kernel.

Reply:
Fixed